### PR TITLE
feat: handle unauthorized error to redirect to login

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](https://github.com/OsmosysSoftware/osmo-x/blob/main/CONTRIBUTING.md#-submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [ ] I have performed unit testing for the new feature added or updated to ensure the new features added are working as expected.
- [x] I have performed preliminary testing to ensure that any existing features are not impacted and any new features are working as expected as a whole.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login page`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional, may include unit testing screenshots as well)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

_Note: Reviewer should ensure that the checklist and description have been populated and followed correctly, and the PR should be merged only after resolving all conversations and verifying that CI checks pass._

---

**Description:**

Handle GraphQL response errors. Logout when `unauthorized` error is encountered

**Related changes:**

- Add `GraphQLFormattedError` object in `ApplicationResponse`
- Add code to handle GraphQL response error
- Create `logoutUser` service function
- Logout when `unauthorized` error is encountered
- Test on local and docker environment

**Screenshots:**

Unauthorized token
https://github.com/user-attachments/assets/f41325b0-690d-4f78-8cf7-699576b2cafb

Non admin user login
https://github.com/user-attachments/assets/29b1e4ba-9fce-402a-9db2-20b12164634f

**Pending actions:**
NA

**Additional notes:**
NA

## Summary by CodeRabbit

- **New Features**
	- Introduced a logout functionality that navigates users to the login page.
	- Enhanced error handling in the applications service and notifications component, specifically addressing unauthorized access.

- **Bug Fixes**
	- Improved error handling logic to manage unauthorized access and log errors effectively.

- **Documentation**
	- Updated interfaces to include error handling properties for better response management.

### CODE REVIEW

File: apps/portal/src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -38,4 +42,10 @@ export class AuthService {
     const variables = { username: data.username, password: data.password };
     return this.graphqlService.mutate(LoginUser, variables);
   }
+
+  logoutUser(): void {
+    // Clear storage and navigate to login
+    localStorage.clear();
+    this.router.navigate(['/login']);
+  }
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/portal/src/app/auth/auth.service.ts:50]: @kshitij-k-osmosys  Check this.

File: apps/portal/src/app/views/applications/applications.service.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ export class ApplicationsService {
           total: response.data?.applications.total,
           offset: response.data?.applications.offset,
           limit: response.data?.applications.limit,
+          errors: response.errors || null,
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/portal/src/app/views/applications/applications.service.ts:32]: @kshitij-k-osmosys  Don't work on this now, once we have proper portal we will be using translate service anyway.

File: apps/portal/src/app/views/notifications/notifications.component.ts [hunk 3]
```diff
@@ -121,7 +123,33 @@ export class NotificationsComponent implements OnInit {
...
+            applicationResponse.errors.forEach((error) => {
+              this.messageService.add({
+                key: 'tst',
+                severity: 'error',
+                summary: 'Error',
+                detail: `GraphQL Error - Get Applications: ${error.message}`,
+              });
+            });
+          }
+        } else if (applicationResponse && applicationResponse.applications) {
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/portal/src/app/views/notifications/notifications.component.ts:152]: @kshitij-k-osmosys  Check if this is a valid suggestion regarding the checks.

File: apps/portal/src/app/auth/auth.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/auth/auth.service.ts]: Added 9c913fbdec92ca85ecb0d6d62726343d8ff65083

File: apps/portal/src/app/views/notifications/notifications.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts]: Added a fix here 9c913fbdec92ca85ecb0d6d62726343d8ff65083

**Review by xixas — APPROVED** [review]

**Pull request merged**